src/views/Register/NameForm.jsx

```diff
@@ -6,28 +6,31 @@ import { useEffect } from "react";
 
 const memberSchema = yup.object().shape({
     teamName: yup.string().required("Team Name is required."),
+    count: yup.number().required("Please enter the team size.").min(3, "Team size needs to be 3 or 4.").max(4)
 }); 
 
 
-function NameForm({ formKey, handleSubmitFunc }) {
-    const { register, handleSubmit, formState: { errors }, reset } = useForm({
+function NameForm({ formKey, handleSubmitFunc, width }) {
+    const { register, handleSubmit, formState: { errors } } = useForm({
         resolver: yupResolver(memberSchema)
     });
 
     useEffect(() => {
         handleSubmitFunc && handleSubmitFunc(formKey, () => {
             return new Promise((resolve, reject) => {
                 handleSubmit((data) => {
-                    resolve(data.teamName);
+                    resolve(data);
                 }, () => {
                     resolve(null);
                 })();
             })
         });
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Remove eslint comments

Reply:
There's a bit issue here. I need this bit of code to run only once in the function, but not use all the vairables in the dependency array, it I do that, it leads to a infinity loop. I wonder why I can't just pass the empty array [] and not see the lint error

Reply:
I removed them anyway